### PR TITLE
extract reward-info crate

### PR DESCRIPTION
**Pull request by kevinheavey**
#### Problem
`solana_sdk::reward_type` and `solana_sdk::reward_info` impose a `solana-sdk` dependency on `solana-rpc-client-api`

#### Summary of Changes
- Move them into a single `solana-reward-info` crate.
- Re-export in their old modules and add deprecation notices
- Make serde optional in the new crate

### CODE REVIEW

**Review by buffalojoec — COMMENTED** [review]
Maybe I'm missing something, but where are these types used in `solana-rpc-client-api`?

As a side note, it looks like everything `solana-rpc-client-api` depends on from the SDK has been broken up into smaller crates, except for these two:

* `SignerError`
* `transaction::Result`

Seems like we're pretty close to being able to just swap in the smaller crates for all of `solana-rpc-client-api`.

**Comment by kevinheavey** [comment]
> Maybe I'm missing something, but where are these types used in `solana-rpc-client-api`?

The dependency is indirect, via solana-transaction-status-client-types

> Seems like we're pretty close to being able to just swap in the smaller crates for all of solana-rpc-client-api.

Yeah this is being tracked in #3027

File: sdk/src/lib.rs [hunk 1]
```diff
@@ -141,7 +143,7 @@ pub use solana_program_memory as program_memory;
 /// assert_eq!(ID, my_id);
 /// ```
 pub use solana_pubkey::pubkey;
-#[deprecated(since = "2.1.0", note = "Use `solana-sanitize` crate instead")]
+#[deprecated(since = "2.2.0", note = "Use `solana-sanitize` crate instead")]
```
Comment (buffalojoec) [inline sdk/src/lib.rs]: Why change this one?

File: sdk/src/reward_info.rs [hunk 2]
```diff
@@ -1,13 +1 @@
...
-pub struct RewardInfo {
-    pub reward_type: RewardType,
-    /// Reward amount
-    pub lamports: i64,
-    /// Account balance in lamports after `lamports` was applied
-    pub post_balance: u64,
-    /// Vote account commission when the reward was credited, only present for voting and staking rewards
-    pub commission: Option<u8>,
-}
+pub use solana_reward_info::RewardInfo;
```
Comment (buffalojoec) [inline sdk/src/reward_info.rs]: Haha, this kind of stinks. Can we just use a `mod` in the SDK's `lib.rs`, so we can cut these files?

```rust
#[deprecated(...)]
pub mod reward_info {
    pub use solana_reward_info::RewardInfo;
}
```

**Review by buffalojoec — COMMENTED** [review]
Works for me. Just 2 nits.

File: sdk/src/reward_info.rs (hunk 2 above)
Comment (kevinheavey) [reply sdk/src/reward_info.rs]: sure

File: sdk/src/lib.rs [hunk 3]
```diff
@@ -88,7 +88,11 @@ pub mod quic;
 pub mod rent_collector;
 pub mod rent_debits;
 pub mod reserved_account_keys;
-pub mod reward_info;
+#[deprecated(since = "2.1.0", note = "Use `solana-reward-info` crate instead")]
+pub mod reward_info {
+    pub use solana_reward_info::RewardInfo;
+}
+#[deprecated(since = "2.1.0", note = "Use `solana-reward-info` crate instead")]
```
Comment (buffalojoec) [inline sdk/src/lib.rs]: Sorry, I should have specified to do this for both.

**Review by buffalojoec — APPROVED** [review]

**Pull request merged**